xymol/xymol.py

```diff
@@ -133,3 +135,43 @@ def plot_similarity_map(
             self.mol, weights, **kwargs
         )
         img.savefig(file_name, bbox_inches='tight', dpi=600)
+
+    def create_map(
+        self, featurizer, model, filename="similarity_map.png"
+        ) -> list:
+        """Create similarity map
+
+        Args:
+            featurizer (class | str): if of type class, featurizer must have a featurize() function.
+            model (class | str): if of type class, model must have a predict() function.
+
+        Returns:
+            list of weights
+        """
+
+        if isinstance(featurizer, str):
```

Comment:
We can just check if the input featurizer class/function has `featurize` API. It's the user's job to have this API.

Reply:
And given in this case, we do not have to provide deepchem featurizers.